### PR TITLE
remove swiftSettings flag for unsafeFlags

### PR DESCRIPTION
**Pull request by tomislaveric**
[ISSUE](https://github.com/antoniusnaumann/cargo-swift/issues/43)

With this PR you will be able to build your iOS/macOS project on newer versions.

### CODE REVIEW

**Comment by antoniusnaumann** [comment]
Does this flag silence warnings in the generated code? If so this is a good fix.

**Comment by tomislaveric** [comment]
Screenshot 2023-12-19 at 16 08 16
if you mean warnings in Xcode, i don't see any so far. But my current setup is very plain. For this example in the image i only have one async func.

**Comment by antoniusnaumann** [comment]
Okay, I'll check it later and will merge afterwards. Thank you for your contribution!

File: templates/Package.swift [hunk 1]
```diff
@@ -25,7 +25,7 @@ let package = Package(
                 .target(name: "RustFramework")
             ]{% if disable_warnings %},
             swiftSettings: [
-                .unsafeFlags(["-suppress-warnings"]),
+                .define("SWIFT_PACKAGE_MANAGER"),
```
Comment (antoniusnaumann) [inline templates/Package.swift:28]: Could you link a source that explains what this define does? A quick search did not yield meaningful results. Thanks!

Comment (tomislaveric) [reply templates/Package.swift:28]: To be honest, i found this somewhere as a quick fix. The only meaningful source i could found for it is the source code itself (see below). However, on my current setup i removee the whole `swiftSettings` flag and the code builds successfully without warnings, so maybe its unnecessary. 

[Here i found the first report ](https://forums.swift.org/t/override-for-unsafeflags-in-swift-package-manager/45273/4) of the issue, with a workaround which sounds like a hack. If you know another way of getting the build succeed please let me know! For now i remove the `swiftSettings` flag, reset the Packages Cache in Xcode and then it builds. 

`/// The following example shows how to use build setting conditions with various
/// APIs:
///
/// ```swift
/// ...
/// .target(
///     name: "MyTool",
///     dependencies: ["Utility"],
///     cSettings: [
///         .headerSearchPath("path/relative/to/my/target"),
///         .define("DISABLE_SOMETHING", .when(platforms: [.iOS], configuration: .release)),
///     ],
///     **swiftSettings: [
///         .define("ENABLE_SOMETHING", .when(configuration: .release)),
///     ],**
///     linkerSettings: [
///         .linkLibrary("openssl", .when(platforms: [.linux])),
///     ]
/// ),
/// ````

Comment (antoniusnaumann) [reply templates/Package.swift:28]: Ok, I think I will apply the solution currently present on main then (not using the suppressWarnings flag by default and include an option that suppresses them)

**Pull request closed without merging**